airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/FeatureFlagFetchActivityImpl.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.ConnectionApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.featureflag.FeatureFlagClient;
+import io.airbyte.featureflag.FieldSelectionEnabled;
+import io.airbyte.featureflag.Flag;
+import io.airbyte.featureflag.Workspace;
+import jakarta.inject.Singleton;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class FeatureFlagFetchActivityImpl implements FeatureFlagFetchActivity {
+
+  private final SourceApi sourceApi;
+  private final ConnectionApi connectionApi;
+  private final FeatureFlagClient featureFlagClient;
+
+  public FeatureFlagFetchActivityImpl(SourceApi sourceApi,
+                                      ConnectionApi connectionApi,
+                                      FeatureFlagClient featureFlagClient) {
+    this.sourceApi = sourceApi;
+    this.connectionApi = connectionApi;
+    this.featureFlagClient = featureFlagClient;
+  }
+
+  public UUID getWorkspaceId(final UUID connectionId) {
+    final ConnectionRead connection = AirbyteApiClient.retryWithJitter(
+        () -> connectionApi.getConnection(new ConnectionIdRequestBody().connectionId(connectionId)),
+        "get connection");
+
+    final SourceRead source = AirbyteApiClient.retryWithJitter(
+        () -> sourceApi.getSource(new SourceIdRequestBody().sourceId(connection.getSourceId())),
+        "get source");
+
+    return source.getWorkspaceId();
+  }
```

Comment:
This was the way I found for how to get the workspace id for a connection via the API. Making two api calls (connection -> source) feels a bit weird, any better way to do this?

Reply:
Sweet, changed in [1d7edb8](https://github.com/airbytehq/airbyte/pull/22960/commits/1d7edb80c4bc7cb6ba5aa7b10f76e68888d305b0)